add string for note-editor a11y link popup message

abaevbog opened the pull request:
String read out by screen readers when the link popup appears instructing users to navigate it.

Needed for https://github.com/zotero/note-editor/pull/65
Addresses: #5199

Not sure if the message needs to be more detailed (e.g. "Use Shift-Tab to navigate it to remove/edit/open the link") or if it's better to just be concise.

dstillman commented:
Do we not yet have a way to use FTL strings in note-editor?

abaevbog commented:
I don't believe so... Fluent is not used anywhere in `note-editor` currently, and if I do `document.getElementById("editor-view").contentDocument.l10n` when a note is opened, `null` is returned. That is in contract to top-level `document.l10n` or `document.querySelector(".reader").contentDocument.l10n`, which both return a fulfilled promise of `DocumentL10n`.

dstillman commented:
I think reader is not remote (currently, though we need to change that), so direct access is possible there. For note-editor, we might need to set up async message-passing to get the strings. If note-editor expects to have strings synchronously, that could be a little ugly.

Or we could have a dedicated .ftl file, make that available to the editor, and use https://github.com/projectfluent/fluent.js/, but that's almost certainly more complicated.

We can merge this for now, but @mrtcode, would be good to think about switching to .ftl, since I imagine we'll need to at some point.

abaevbog commented:
Oh, wait a second. If I add `<link rel="localization" href="zotero.ftl" />` to the `editor.zotero.html`, it actually does seem to work! I may have been looking at the wrong `.html` file from `note-editor` before. So I guess we would want to do that instead, and then fetch the correct string via `l10n.formatValue`?

abaevbog commented:
Well, actually though, I'm not sure how this fluent approach would work on other platforms? We could do a check similar to what we do in the reader, where if `document.l10n` exists, we set the message, and otherwise do nothing - at least for now. It would likely not be relevant for ios and android, so it's just about how web library would be able to handle the `document.l10n`.

dstillman commented:
And without that, it just uses the bundled English strings, right? So we'll need to do something different to localize the reader and editor on other platforms, but that's it.

abaevbog commented:
Yes, without that (as it does in the current state of the linked PR), it uses `intl.formatMessage`, which fetches bundled `zotero.properties` strings, at least on desktop.

So we could do something like that? Use `document.l10n` if it's there, or default to `intl.formatMessage` otherwise.
```
{ document.l10n ? 
    <div id="a11y-alert" aria-live="polite" data-l10n-id={editorState.link.popup.active ? "note-editor-a11y-link-popup-appearedalert" : ""}></div>
    : <div id="a11y-alert" aria-live="polite">{editorState.link.popup.active ? intl.formatMessage({ id: 'noteEditor.a11yLinkPopupAppearedAlert' }) : ""}</div>
}
```

dstillman commented:
Sorry, I mean the duplicate bundled English strings in the reader (and editor?) submodule. We're not using localization at all in the other environments.

mrtcode commented:
> I think reader is not remote (currently, though we need to change that)

We need to be sure it's worth doing this, because the PDF view uses a worker for part of the processing, and I believe the EPUB view does something similar—possibly using a separate iframe (@AbeJellinek can explain this better). Losing synchronous access to the reader would be quite inconvenient for both us and plugin developers.

Additionally, memory usage is a bigger concern.

abaevbog commented:
> Sorry, I mean the duplicate bundled English strings in the reader (and editor?) submodule. We're not using localization at all in the other environments, so it needs those strings in order to show anything.

Ah, right! I was a bit confused about this mechanism. Now I see that in the desktop app, localized strings are [passed](https://github.com/zotero/zotero/blob/a7778b93e841ce62773efc65cd86576c7bfe8af1/chrome/content/zotero/xpcom/editorInstance.js#L197) from `zotero.properties`, and `en-us.strings.js` is used for all other platforms (e.g. [web](https://github.com/zotero/note-editor/blob/ae92348364daec4b6877e6805b54a5cbcf145741/src/index.web.js#L7)). 

> So I'm saying if we can use Fluent in Zotero and continue using the in-submodule strings otherwise for now, that's fine.

Great. I updated the `note-editor` PR to use fluent in the desktop app and fetch strings via `document.l10n.formatValue`. If `document.l10n` is not available, we'll fetch strings from `en-us.strings.js` via `intl.formatMessage`, the way it's done currently. I moved all of this into a new `A11yAlert` component, to not clutter `src/ui/editor.js`.
